assets/src/design-system/components/typeahead/input/components.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+/**
+ * Internal dependencies
+ */
+import { Chevron, Close as Clear, Search } from '../../../icons';
+import { themeHelpers } from '../../../theme';
+
+export const InputContainer = styled.div`
+  display: flex;
+  position: relative;
+  box-sizing: border-box;
+  height: 36px;
+  background-color: ${({ theme }) => theme.colors.bg.primary};
+`;
+
+export const Input = styled.input(
+  ({ theme, isOpen, hasError, hasSearchIcon }) => css`
+    position: absolute;
+    padding: 8px 20px 8px 12px;
+    padding-left: ${hasSearchIcon ? 32 : 12}px;
+    height: 100%;
+    width: 100%;
+    background-color: transparent;
+    outline: none;
+    color: ${theme.colors.fg.primary};
+    border-radius: ${theme.borders.radius.small};
+    ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)};
+
+    ${themeHelpers.expandPresetStyles({
+      preset: theme.typography.presets.paragraph.small,
+      theme,
+    })};
+
+    /* border color needs to be reapplied from helper for different states */
+    border-color: ${theme.colors.border[
+      isOpen ? 'defaultActive' : 'defaultNormal'
+    ]};
+
+    &::-ms-clear {
+      display: none;
+    }
+
+    &::-webkit-search-decoration,
+    &::-webkit-search-cancel-button,
+    &::-webkit-search-results-button,
+    &::-webkit-search-results-decoration {
+      appearance: none;
+    }
+
+    &:hover {
+      border-color: ${theme.colors.border[
+        isOpen ? 'defaultActive' : 'defaultHover'
+      ]};
+    }
```

Comment:
Not sure if this'll cause a linting error tho 🤷 
```suggestion
    &::-ms-clear {
      display: none;
    }

    &::-webkit-search-decoration,
    &::-webkit-search-cancel-button,
    &::-webkit-search-results-button,
    &::-webkit-search-results-decoration {
      appearance: none;
    }
    
    &,
    &:hover {
      border-color: ${theme.colors.border[
        isOpen ? 'defaultActive' : 'defaultHover'
      ]};
    }
```

Reply:
this would be nice - the linter doesn't like it though (plus it loses `defaultNormal` as the default border color. Lots of borders here).